lib/matplotlib/table.py

```diff
@@ -744,6 +746,13 @@ def table(ax,
         cols = len(cellColours[0])
         cellText = [[''] * cols] * rows
 
+    # Check if we have a Pandas DataFrame
+    if _is_pandas_dataframe(cellText):
+        # Convert to numpy array
+        header = _unpack_to_numpy(cellText.columns)
+        data = _unpack_to_numpy(cellText)
```

Comment:
You don't have to go through the generic `_unpack_to_numpy`, because you now know that you have a DataFrame and can work on it:
```suggestion
        header = cellText.columns.to_numpy()
        data = cellText.to_numpy()
```